targets/TARGET_Samsung/security_subsystem/api/trng_api.c

```diff
@@ -0,0 +1,54 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include "hal/trng_api.h"
+#include "sss_driver_rng.h"
+#include "mbed_error.h"
+
+void trng_init(trng_t *obj)
+{
+    (void)obj;
+}
+
+void trng_free(trng_t *obj)
+{
+    (void)obj;
+}
+
+int trng_get_bytes(trng_t *obj, uint8_t *output, size_t length, size_t *output_length)
+{
+    (void)obj;
+
+    stOCTET_STRING stRANDOM;
+    unsigned int ret = 0;
+
+    stRANDOM.pu08Data = (u08 *)output;
+    ret = sss_generate_rawrandom(&stRANDOM, length);
```

Comment:
How many bits of entropy will be contained in the output buffer per bit? For example some TRNG hardware might theoretically place 8 bytes of entropy within every 32 bytes of output, for an estimate entropy content of 0.25 bits per bit.

Mbed OS assumes that trng_get_bytes() will return 1.00 bits of estimated entropy per bit content. With any lesser estimate of entropy, the security of an Mbed OS system is lessened.

Reply:
The function returns an an approximation of 1.

Reply:
@Patater. Please let us know if this doesn't answer your question.